packages/python/pyfora/src/core/PyObjectPtr.hpp

```diff
@@ -0,0 +1,232 @@
+/***************************************************************************
+   Copyright 2016 Ufora Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+****************************************************************************/
+#pragma once
+
+#include <Python.h>
+
+#include <cstddef>
+
+
+class PyObjectPtr {
+public:
+    PyObjectPtr(PyObject* pyObject)
```

Comment:
This suggests implicit conversion between PyObject* and PyObjectPtr.
Is that intentional? If you do want to allow implicit conversion, it may be safer to make the conversion constructor incref the pointer.
Otherwise, use the 'explicit' keyword to make the conversion explicit.
In general, it seems a bit dangerous to allow the creation of a PyObjectPtr without increfing because you're still going to decref in the destructor.